prelude/toolchains/msvc/vswhere.py

```diff
@@ -202,6 +202,76 @@ def write_tool_json(out, tool):
     out.write(j)
 
 
+# for use with the ewdk to grab the environment strings
+def get_env(ewdkdir: str):
+    """
+    Inspiration taken from the following:
+    http://pythonwise.blogspot.fr/2010/04/sourcing-shell-script.html (Miki Tebeka)
+    http://stackoverflow.com/questions/3503719/#comment28061110_3505826 (ahal)
+    """
+
+    # We need to write the script that will make the important variables available
+    # This gets run from the temporary build directory, so this does not pollute the sources
+    vcVarsExtract = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                "VcVarsExtract.bat")
+    if not os.path.exists(vcVarsExtract):
+        with open(vcVarsExtract, "w") as f:
+            f.write("@echo off\r\ncall %%1 %%2 %%3 %%4 %%5 %%6 %%7 %%8 %%9 > NUL\r\nset")
```

Comment:
You shouldn't create files next to the target inputs. Buck should set temporary directory for you.
```suggestion
    with tempfile.NamedTemporaryFile(prefix="VcVarsExtract", suffix=".bat") as tmp:
        print("@echo off", file=tmp)
        print("call %* > NUL", file=tmp)
        print("env", file=tmp)
```
```

Reply:
Most of this is fine, but the final `set` can't be changed to `env`, because the `\` are imperfectly escaped. I also had to `delete=False`

Reply:
Sure, let's keep `set` then.